copy-conditions.go

```diff
@@ -91,9 +75,37 @@ func (c *CopyConditions) SetModified(modTime time.Time) error {
 	if modTime.IsZero() {
 		return ErrInvalidArgument("Modified since cannot be empty.")
 	}
-	c.conditions = append(c.conditions, copyCondition{
-		key:   "x-amz-copy-source-if-modified-since",
-		value: modTime.Format(http.TimeFormat),
-	})
+	c.conditions["x-amz-copy-source-if-modified-since"] = modTime.Format(http.TimeFormat)
 	return nil
 }
+
+// SetByteRange - set the start and end of the source object to be
+// copied.
+func (c *CopyConditions) SetByteRange(start, end int64) error {
+	if start < 0 || end < start {
+		return ErrInvalidArgument("Range start less than 0 or range end less than range start.")
+	}
+	if end-start+1 < 1 {
```

Comment:
I think this condition is useless, 

Here we are already sure that `start <= end`:

`start <= end` **<=>** `0 <= end - start` **<=>** `1 <= end - start + 1`, 

so we will never have `end-start+1 < 1`